clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -8822,6 +8842,9 @@ void Sema::ProcessDeclAttributeList(Scope *S, Decl *D,
     } else if (const auto *A = D->getAttr<IntelReqdSubGroupSizeAttr>()) {
       Diag(D->getLocation(), diag::err_opencl_kernel_attr) << A;
       D->setInvalidDecl();
+    } else if (const auto *A = D->getAttr<SYCLIntelStallEnableAttr>()) {
+      Diag(D->getLocation(), diag::err_opencl_kernel_attr) << A;
+      D->setInvalidDecl();
```

Comment:
These lines seem not necessary. The attribute `stall_enable` seems to be enabled only for SYCL and this section of the code produces errors for attributes which should be applied to OpenCL kernel in OpenCL language and all these diagnostics are disabled for SYCL. See if statement on lines 8822-8823 in this file.
```suggestion
```

Reply:
> It looks ok in general except my little comment and the fact that I completely didn't get the doc, because it looks like very specific for some devices (FPGA?).

I meant that, I didn't get the doc because I'm not familiar with FPGA devices. Does it make sense to mention that it is some FPGA-specific thing?

Reply:
i looked at other new FPGA function attribute "SYCLIntelSchedulerTargetFmaxMhz" . It did not add those line (i think i can remove those).

i do not see anywhere it is mentioned to be a FPGA specific.

Reply:
Previously we had FPGA specific attributes in `intelfpga` namespace, so it was pretty much obvious to which device they belong. Now when `intelfpga` was deprecated in favor of just `intel` namespace we can re-consider texts in some warnings etc, but it can be done in a separate patch.

Reply:
I have created separate PR for this issue. 

https://github.com/intel/llvm/issues/2740